--- a/arch/arm/mach-mediatek/mt7622/init.c
+++ b/arch/arm/mach-mediatek/mt7622/init.c
@@ -32,10 +32,12 @@ int dram_init(void)
 	return 0;
 }
 
+#if !defined(CONFIG_SYSRESET_PSCI)
 void reset_cpu(void)
 {
 	psci_system_reset();
 }
+#endif
 
 static struct mm_region mt7622_mem_map[] = {
 	{
--- a/arch/arm/mach-mediatek/mt7981/init.c
+++ b/arch/arm/mach-mediatek/mt7981/init.c
@@ -26,10 +26,12 @@ int dram_init(void)
 	return 0;
 }
 
+#if !defined(CONFIG_SYSRESET_PSCI)
 void reset_cpu(void)
 {
 	psci_system_reset();
 }
+#endif
 
 static struct mm_region mt7981_mem_map[] = {
 	{
--- a/arch/arm/mach-mediatek/mt7986/init.c
+++ b/arch/arm/mach-mediatek/mt7986/init.c
@@ -26,10 +26,12 @@ int dram_init(void)
 	return 0;
 }
 
+#if !defined(CONFIG_SYSRESET_PSCI)
 void reset_cpu(void)
 {
 	psci_system_reset();
 }
+#endif
 
 static struct mm_region mt7986_mem_map[] = {
 	{
--- a/arch/arm/mach-mediatek/mt7988/init.c
+++ b/arch/arm/mach-mediatek/mt7988/init.c
@@ -34,10 +34,12 @@ int dram_init_banksize(void)
 	return 0;
 }
 
+#if !defined(CONFIG_SYSRESET_PSCI)
 void reset_cpu(ulong addr)
 {
 	psci_system_reset();
 }
+#endif
 
 static struct mm_region mt7988_mem_map[] = {
 	{
